### PR TITLE
[#166761737] Order the list of organizations in alphabetical order

### PR DESCRIPTION
**Pull request by Alex193a**
In the list of services, the institutions are now sorted in alphabetical order.

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#166761737](https://www.pivotaltracker.com/story/show/166761737): Come CIT nella lista servizi voglio vedere gli enti in ordine alfabetico

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/components/services/ServiceSectionListComponent.tsx [hunk 1]
```diff
@@ -76,6 +76,8 @@ class ServiceSectionListComponent extends React.Component<Props> {
       <RefreshControl refreshing={isRefreshing} onRefresh={onRefresh} />
     );
 
+    sections.sort((a, b) => (a.title || "").localeCompare(b.title));
```
Comment (Undermaken) [inline ts/components/services/ServiceSectionListComponent.tsx]: - avoid object mutation
- sections sorting is done on each state/props update (you are in render function) causing low performance and frames drop (section could have several items) . Try instead to change the caller component **ServicesScreen**: create a selector to get all sections by replacing the existing one https://github.com/teamdigitale/io-app/blob/548fff7b0803c4b313276d0695afc12f798476ff/ts/screens/preferences/ServicesScreen.tsx#L128-L140 keep sections into the component's state and check when it must be sorted again (_getDerivedStateFromProps_ could be useful)

Comment (Alex193a) [reply ts/components/services/ServiceSectionListComponent.tsx]: @Undermaken can you check the latest [commit](https://github.com/teamdigitale/io-app/pull/1153/commits/9c70888f4ccac64de7e2133b8bad26d264be92ec), please? Should it be done in that way?

Comment (Undermaken) [reply ts/components/services/ServiceSectionListComponent.tsx]: @Alex193a sorry for the delay. I will be off until next Tuesday. I will review asap. Until that keep this PR sync with master. Thank yoi ;)

File: ts/store/reducers/entities/services/orderedServices.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { createSelector } from "reselect";
```
[the remark this answers is not in the document]
Comment (Alex193a) [reply ts/store/reducers/entities/services/orderedServices.ts]: I've moved the selector to OldServicesHomeScreen.tsx

File: ts/screens/preferences/OldServicesHomeScreen.tsx [hunk 3]
```diff
@@ -124,25 +125,33 @@ class OldServicesHomeScreen extends React.Component<Props> {
...
+        const serviceIdsForOrg = services.byOrgFiscalCode[fiscalCode] || [];
+        const data = serviceIdsForOrg
+          .map(id => services.byId[id])
+          .filter(isDefined);
+        return {
+          title,
+          data
+        };
+      })
+      .sort((a, b) => (a.title || "").localeCompare(b.title))
```
Comment (Undermaken) [inline ts/screens/preferences/OldServicesHomeScreen.tsx]: @Alex193a  You could sort the array just after filtered it, just an optimization ;)

Comment (Alex193a) [reply ts/screens/preferences/OldServicesHomeScreen.tsx]: Done! [0c97f3c](https://github.com/teamdigitale/io-app/pull/1153/commits/0c97f3ceeee8cb4bfd1d4f560c78ee1457efa7b2)

**Pull request merged**